### ISSUE TITLE
unzip_requirements leaves /tmp in a bad state

### ISSUE THREAD

**Issue by cpb8010** [UnitedIncome/serverless-python-requirements#151]
Hello!  🎈 

The symptom of the problem I'm seeing is that on the initial boot of Lamba, some modules will fail to load due to missing files (different modules and files most times). The hot instance is then unusable until it's restarted. This happens randomly on a service with maybe one or two concurrent instances with a very low request load, producing failures about once a week.

I thought I was able to trace this down before to a requirements archive that would inflate to just short of the 500 MB /tmp mount limit without throwing disk space errors.
Reducing the uncompressed size of the archive has reduced the frequency of failures, but this issue persists with an archive that's well within the variability of the extracted package.

I don't think my python-requirements or serverless setup are unreasonable, but I'll post them here for good measure.

package
```
"serverless": "^1.26.0"
"serverless-python-requirements": "^3.1.1",
```

python requirements config: 
```
pythonRequirements:
    fileName: requirements.txt
    dockerizePip: non-linux
    zip: true
```

requirements.txt
```
dill==0.2.7.1
numpy==1.13.1
pandas==0.20.3
PyMySQL==0.7.11
python-dateutil==2.6.1
pytz==2017.2
raven==6.5.0
raven-python-lambda==0.1.4
scipy==0.19.1
scikit-learn==0.19.1
```

The whole zip file is 114.95 MB compressed. The .requirements.zip inflated is anywhere from 396 MB (extracted locally) to 416 MB (measured on the lambda). I also can't explain the reason for the large size discrepancy between my local (OSX) file system and AWS. 
Even worse, my space usage measurements for unzipping fluctuate (5%) without impacting the functionality of the service.

----
**TL;DR**

What can be done to improve the reliability of the unzipping process?
Or this an AWS problem?

Thanks!

**Comment by dschep** [comment]
That should be addressed by #150, which I merged yesterday. It'll be in the `4.0.0` release. I'll probably cut a `4.0.0-beta.2` soon.
